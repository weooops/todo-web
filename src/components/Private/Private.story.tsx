import * as React from 'react';
import { Provider } from 'react-redux';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import configureStore from '../../configureStore';
import Container from './container';

const store = configureStore();

const props = {
  todos: [
    { id: 1, title: 'a', message: 'aa' },
    { id: 2, title: 'b', message: 'bb' },
    { id: 3, title: 'c' }
  ],
  getTodos: action('getTodos')
};
const component = <Container {...props} />;

storiesOf('Private', module)
  .addDecorator(story => <Provider store={store}>{story()}</Provider>)
  .add('default', () => component);
