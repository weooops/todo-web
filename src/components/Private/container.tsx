import * as React from 'react';

import { TodoType } from '../../models/';
import Private from './Private';

interface PropsType {
  todos: Array<TodoType>;
  getTodos(): void;
}

interface StateType {
  loading: boolean;
}

class Container extends React.Component<PropsType, StateType> {
  state: StateType = {
    loading: true
  };

  componentDidMount(): void {
    const { todos, getTodos } = this.props;

    if (todos.length) {
      this.setState({ loading: false });
    } else {
      getTodos();
    }
  }

  componentWillReceiveProps(nextProps: PropsType): void {
    if (nextProps.todos.length) {
      this.setState({ loading: false });
    }
  }

  render(): JSX.Element {
    const { todos } = this.props;
    const { loading } = this.state;

    return (
      <Private
        loading={loading}
        todos={todos}
      />
    );
  }
}

export default Container;
