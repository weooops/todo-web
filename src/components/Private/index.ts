import { Dispatch } from 'redux';
import { connect } from 'react-redux';

import { State } from '../../reducers';
import { Action } from '../../actions';
import { getTodos } from '../../actions/todos';
import Container from './container';

const mapStateToProps = (state: State) => ({
  todos: state.todos
});

const mapDispatchToProps = (dispatch: Dispatch<Action>) => ({
  getTodos: () => {
    dispatch(getTodos());
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(Container);
