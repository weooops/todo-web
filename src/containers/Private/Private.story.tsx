import * as React from 'react';
import { Provider } from 'react-redux';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import configureStore from '../../configureStore';
import Container from './container';

const store = configureStore();

const props = {
  todos: [
    { id: 1, title: 'a', message: 'aa', updatedAt: '2018-01-28 06:52:05' },
    { id: 2, title: 'b', message: 'bb', updatedAt: '2018-01-28 06:52:05' },
    { id: 3, title: 'c', message: 'cc', updatedAt: '2018-01-28 06:52:05' }
  ],
  getTodos: action('getTodos')
};
const component = <Container {...props} />;

storiesOf('Private', module)
  .addDecorator(story => <Provider store={store}>{story()}</Provider>)
  .add('default', () => component);
