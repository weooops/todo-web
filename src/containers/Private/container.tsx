import * as React from 'react';

import { TodoType } from '../../models/';
import Private from './Private';

interface IProps {
  todos: Array<TodoType>;
  getTodos(): void;
}

interface IStates {
  loading: boolean;
}

class Container extends React.Component<IProps, IStates> {
  public state: IStates = {
    loading: true
  };

  public componentDidMount(): void {
    const { todos, getTodos } = this.props;

    if (todos.length) {
      this.setState({ loading: false });
    } else {
      getTodos();
    }
  }

  public componentWillReceiveProps(nextProps: IProps): void {
    if (nextProps.todos.length) {
      this.setState({ loading: false });
    }
  }

  public render(): JSX.Element {
    const { todos } = this.props;
    const { loading } = this.state;

    return (
      <Private
        loading={loading}
        todos={todos}
      />
    );
  }
}

export default Container;
