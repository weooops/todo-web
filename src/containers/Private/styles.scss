.private {
  // margin: 2px;
  // border: 2px solid brown;
  max-width: 620px;
  width: 100%;
}
