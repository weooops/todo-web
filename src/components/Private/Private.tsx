import * as React from 'react';

import * as styles from './styles.scss';
import * as classNames from 'classnames/bind';
import { TodoType } from '../../models';
import Loading from '../Loading';
import TodoList from '../TodoList';

const cx = classNames.bind(styles);

export interface PropsType {
  todos: Array<TodoType>;
  loading: boolean;
}

const App: React.SFC<PropsType> = props => {
  if (props.loading) {
    return <Loading />;
  }
  return <Render {...props} />;
};

const Render: React.SFC<PropsType> = props => (
  <div className={cx('unauth')}>
    <h2>TODO 목록</h2>
    <div>
      <TodoList todos={props.todos} />
    </div>
  </div>
);

export default App;
