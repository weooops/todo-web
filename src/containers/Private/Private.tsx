import * as React from 'react';

import * as styles from './styles.scss';
import * as classNames from 'classnames/bind';
import { TodoType } from '../../models';
import Loading from '../../components/Loading';
import TodoList from '../../components/TodoList';

const cx = classNames.bind(styles);

interface IProps {
  todos: Array<TodoType>;
  loading: boolean;
}

const App: React.SFC<IProps> = props => {
  if (props.loading) {
    return <Loading />;
  }
  return <Render {...props} />;
};

const Render: React.SFC<IProps> = props => (
  <div className={cx('unauth')}>
    <h2>TODO 목록</h2>
    <div>
      <TodoList todos={props.todos} />
    </div>
  </div>
);

export default App;
