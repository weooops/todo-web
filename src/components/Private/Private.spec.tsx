import * as React from 'react';
import { shallow } from 'enzyme';

import Private from './Private';

it('renders correctly', () => {
  const props = {
    loading: false,
    todos: [
      { id: 1, title: 'a', message: 'aa' },
      { id: 2, title: 'b', message: 'bb' },
      { id: 3, title: 'c' }
    ]
  };
  const wrapper = shallow(<Private {...props} />);
  expect(wrapper).toMatchSnapshot();
});
