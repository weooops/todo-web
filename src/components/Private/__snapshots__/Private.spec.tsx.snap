// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    loading={false}
    todos={
      Array [
        Object {
          "id": 1,
          "message": "aa",
          "title": "a",
        },
        Object {
          "id": 2,
          "message": "bb",
          "title": "b",
        },
        Object {
          "id": 3,
          "title": "c",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "loading": false,
      "todos": Array [
        Object {
          "id": 1,
          "message": "aa",
          "title": "a",
        },
        Object {
          "id": 2,
          "message": "bb",
          "title": "b",
        },
        Object {
          "id": 3,
          "title": "c",
        },
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "loading": false,
        "todos": Array [
          Object {
            "id": 1,
            "message": "aa",
            "title": "a",
          },
          Object {
            "id": 2,
            "message": "bb",
            "title": "b",
          },
          Object {
            "id": 3,
            "title": "c",
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
